### PR TITLE
ListRooms using `sid`

### PR DESCRIPTION
**Pull request by boks1971**

### CODE REVIEW

**Review by cnderrauber — APPROVED** [review]

File: pkg/service/redisstore.go [hunk 1]
```diff
@@ -76,37 +77,27 @@ func (s *RedisStore) LoadRoom(_ context.Context, name livekit.RoomName) (*liveki
...
 	for _, item := range items {
 		room := livekit.Room{}
 		err := proto.Unmarshal([]byte(item), &room)
 		if err != nil {
 			return nil, err
 		}
-		rooms = append(rooms, &room)
+
+		if (names == nil && sids == nil) || funk.Contains(names, livekit.RoomName(room.Name)) || funk.Contains(sids, livekit.RoomID(room.Sid)) {
+			if !funk.Contains(rooms, &room) {
```
Comment (davidzhao) [inline pkg/service/redisstore.go:97]: don't think this is contains check is needed. each room is a new object that cannot exist in rooms.

File: pkg/service/localstore.go [hunk 2]
```diff
@@ -50,13 +50,15 @@ func (s *LocalStore) LoadRoom(_ context.Context, name livekit.RoomName) (*liveki
...
-func (s *LocalStore) ListRooms(_ context.Context, names []livekit.RoomName) ([]*livekit.Room, error) {
+func (s *LocalStore) ListRooms(_ context.Context, names []livekit.RoomName, sids []livekit.RoomID) ([]*livekit.Room, error) {
 	s.lock.RLock()
 	defer s.lock.RUnlock()
 	rooms := make([]*livekit.Room, 0, len(s.rooms))
 	for _, r := range s.rooms {
-		if names == nil || funk.Contains(names, livekit.RoomName(r.Name)) {
-			rooms = append(rooms, r)
+		if (names == nil && sids == nil) || funk.Contains(names, livekit.RoomName(r.Name)) || funk.Contains(sids, livekit.RoomID(r.Sid)) {
+			if !funk.Contains(rooms, r) {
```
Comment (davidzhao) [inline pkg/service/localstore.go:59]: don't think this is needed.

**Review by davidzhao — APPROVED** [review]

File: pkg/service/redisstore.go (hunk 1 above)
Comment (boks1971) [reply pkg/service/redisstore.go:97]: Oh my bad. So, have to check for `sid` match.

File: pkg/service/localstore.go (hunk 2 above)
Comment (boks1971) [reply pkg/service/localstore.go:59]: Same as above. Have to check for `sid` match. Wonder how the overlapping test passed (where `names` and `sids` point to the same room). It returned only one object. Have to check that.

Comment (boks1971) [reply pkg/service/localstore.go:59]: Okay, I am mistaken. It is looping over all rooms and checking for matches. So, yeah, the contains check is not needed. Will remove

**Pull request merged**